remove .NET 3.5 project shims from the public API

blairconrad opened the pull request:
Fixes #565.
Extracted from #564.

adamralph commented on Source/FakeItEasy/Configuration/BuildableCallRule.cs:
```diff
@@ -113,7 +113,7 @@ public void WriteDescriptionOfValidCall(IOutputWriter writer)
         public virtual void ApplyWherePredicate(Func<IFakeObjectCall, bool> predicate, Action<IOutputWriter> descriptionWriter)
         {
-            this.wherePredicates.Add(Tuple.Create(predicate, descriptionWriter));
+            this.wherePredicates.Add(new Tuple<Func<IFakeObjectCall, bool>, Action<IOutputWriter>>(predicate, descriptionWriter));
```
I think `Tuple.Create()` is the idiomatic method of constructing a `Tuple<..>`. It may be worth preserving the static shim which provides it.

blairconrad replied:
Blow me down. I should've looked for this. We were using it in two out of three places. Should I, as part of a final cleanup commit, make the 3rd (`ZipEnumerable`) use it as well?

adamralph replied:
> Should I, as part of a final cleanup commit, make the 3rd (ZipEnumerable) use it as well?

:+1:

blairconrad replied:
Also, the remove/add could be considered messy. Do you prefer a complete rework and force push, or just an additional commit?

adamralph replied:
I don't mind a force push. It's often better than keeping a series of commits which all have a single ultimate aim.

adamralph commented on Source/FakeItEasy.Net35/InternalsVisibleTo.cs:
```diff
@@ -0,0 +1,3 @@
+﻿﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("FakeItEasy.Net35.Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001002f50d82092713be482c885861c984334606da0f54c78738a3dd0862fb2dfc6080f0780132cc65d88f0f0c70af74e8a53430962395bfc1a36fab08b7a2549d387e805c13cc84acd884447ec8c4dcfb6216df720f0998380f9c906b5de8141798d64661f036d47274e6ecb76c9cde5f4cf2b521040601e44b3914fbeb9f39127f9")]
```
No line ending? Tut tut :wink:

blairconrad replied:
Oh? Lots of files don't have line endings. Didn't know we cared. Will add.

adamralph replied:
I imagine we probably have a lot of inconsistency already.

I usually like to add a http://editorconfig.org/ file in the root which enforces it.

blairconrad replied:
The would only work if I had the plugin, no?

blairconrad replied:
Sounds like not such a bad idea. Maybe in a different issue. :wink:

adamralph replied:
Yeah, you need the plug-in. It does indeed sounds like a good candidate for a new issue.

adamralph commented on Source/FakeItEasy.Net35/FakeItEasy.Net35.csproj:
```diff
@@ -693,6 +696,7 @@
       <Link>Properties\FakeItEasy.Dictionary.xml</Link>
     </CodeAnalysisDictionary>
   </ItemGroup>
+  <ItemGroup />
```
Can be reverted.

blairconrad replied:
Man. I've no idea how I managed that. :blush:

adamralph replied:
I believe it's a helpful Visual Studio feature...

blairconrad replied:
Still, I should've noticed in the diff and reverted.

adamralph commented:
@blairconrad changes look good. I left just a couple of very minor comments.

blairconrad commented:
Thanks for the great comments, @adamralph. I think I've addressed everything. I ditched the offending "inline Tuple" commit, reworked the other two, and then added the obligatory cleanup commit which fixed some readonlys and removed some not needed EnumerableExtension stuff, including the copy of `ZipEnumerable`. (Which in the process let me get rid of the extra usage of the `Tuple<T1, T2>` constructor.)

adamralph commented:
:+1: great stuff!

blairconrad commented:
You're kind. Thanks for the constructive comments and merge.
We'll still need a little work to use xUnit 2.0 in the 3.5 unit test package, but it's looking better than it was.
